### ISSUE TITLE
Graphical issues.

### ISSUE THREAD

**Issue by s1eve-mcdichae1** [nxengine/nxengine-evo#260]
Graphics are pushed to the top-left of the screen instead of centered. Some assets appear to be drawn past the edge (perhaps areas that are meant to be "off-screen"?) "Fullscreen: on" doesn't seem to do anything but hide the mouse pointer.

I am running on Raspberry Pi 4 (ARM Linux/Debian 10) under RetroPie (Raspbian/Raspberry Pi OS Lite). My display is 720p.

![20220118_094717](https://user-images.githubusercontent.com/92833764/149998350-1f7a1d33-3b16-4713-8511-5a574113fe26.png)

I have built, prepared, and launched with:

    mkdir build && cd build
    cmake -DCMAKE_BUILD_TYPE=Release -DPORTABLE=On .. && make
    cd ..
    wget https://cavestory.org/downloads/cavestoryen.zip
    unzip cavestoryen.zip -d .
    cp -r CaveStory/data .
    cp CaveStory/Doukutsu.exe .
    wget https://github.com/nxengine/translations/releases/download/v1.14/all.zip
    unzip all.zip -d .
    build/nxextract
    mkdir nxe
    cp build/nxengine-evo nxe
    cp -r data nxe
    mv nxe ~
    cd
    nxe/nxengine-evo

**Comment by isage** [comment]
I'm assuming it's running on framebuffer?
Does game resolution in settings match framebuffer resolution?

**Comment by s1eve-mcdichae1** [comment]
> I'm assuming it's running on framebuffer?

Correct, RetroPie (built on Raspbian Lite) doesn't have a desktop environment, so I think that means it's all terminal and framebuffer graphics.

> Does game resolution in settings match framebuffer resolution?

Not sure, I think it's set to my display resolution (1280x720)? I know that once I get around to launching it from the front-end there are settings I can use to set the resolution per-game, but I'm not sure how this translates to command-line and I'm still trying to get it to work from terminal before I "install" it to the FE.

Do you know how I would set the framebuffer resolution, or would that be platform-dependent and I should go ask the guys at RetroPie?

**Comment by isage** [comment]
Ok, so the game doesn't see framebuffer resolution correctly. That might be an sdl2 bug.
Can you post log file? (Console output or one saved in saves dir ~/.local/share/nxengine)

**Comment by s1eve-mcdichae1** [comment]
I went ahead and installed it to my front-end so I could set the resolution that way.

It looks good at 1024x768 but noticeably sluggish and the animation is a little choppy.

I attribute this to me trying to run high-resolution graphics on a low-power device. It runs better at 640x480, but then the bottom of the image is cut off:

![9B208B18-0859-4072-8CAB-08139891924A](https://user-images.githubusercontent.com/92833764/150033399-6cf5f38b-c0d4-4119-841a-a9e00d3798fc.jpeg)


It's cut off like that on my TV but when I take a framebuffer screenshot with `raspi2png` it shows the whole thing:

![20220118_124534](https://user-images.githubusercontent.com/92833764/150033442-169ea292-3180-4bba-9d45-77e8c5ac5c82.png)


I have previously disabled overscan in my `/boot/config.txt`; if I re-enable it I see the whole image, but then I get additional black borders around everything (terminal, front-end, other games, this game even):

![20220118_150348](https://user-images.githubusercontent.com/92833764/150033485-5738440d-0ad4-4b52-8700-20242f07f231.png)

![20220118_150604](https://user-images.githubusercontent.com/92833764/150033514-d98a5e8d-759f-488b-bca2-8edbf63a7145.png)

**Comment by s1eve-mcdichae1** [comment]
Sorry I was typing out my last reply when yours was posted.

Here are a couple of logs. First one with the default resolution settings: https://pastebin.com/rMCHmwxS

And here's one when I set the front end to launch it in 640x480 (still not sure what that's doing under the hood, but it does change how it looks on screen): https://pastebin.com/eTuL0m27

**Comment by isage** [comment]
Doesn't look like it disables some resolutions, so 720p should be available in game settings

**Comment by s1eve-mcdichae1** [comment]
> Doesn't look like it disables some resolutions, so 720p should be available in game settings

With the default screen resolution, or when explicitly setting any of the five "1280x720" modes in my front-end, the in-game settings options are 640x480, 480x272, 1024x600, and 320x240.

When I set the FE to use 1024x768, that option becomes available in-game as well.

When I set the FE to use 1920x1080, a whole bunch of other options appear in-game including a 1920x1080, but *still* no 1280x720.

Oddly, I did find a 640x480 mode that works. If I select the one @75Hz then the bottom is not cut off like it is on all three of the 640x480@60Hz options did, and it runs noticeably faster this way, on my ARM device, than at those higher resolutions.

So I guess *I'm* done with this issue, since I don't need to use a 1280x720 resolution, but I guess someone else still might.
